minvws/nl-kat-coordination
# feat: adds notification styling and icons

### Changes

- Adds notification icons
- Adds block notification styling, including button styling. Which adds a new button type "plain"

Design: https://www.figma.com/design/MDbpc9K0qjyaoRGAGiDX6t/%F0%9F%8E%A8-KAT-%7C-Design-System-%7C-Components?m=auto&node-id=406-2051&t=t6P0qKKiGMV9HZ5K-1

### Issue link

Closes https://github.com/minvws/nl-kat-coordination/issues/3453

### Demo

**Before** 
"explanation" as example
<img width="993" alt="Screenshot 2024-09-04 at 18 33 49" src="https://github.com/user-attachments/assets/5b436200-ab27-4dd7-8056-f502ace2c02e">

**After**
Explanation: 
<img width="1014" alt="Screenshot 2024-09-04 at 18 34 00" src="https://github.com/user-attachments/assets/310e246e-10cf-4418-8f04-afeb9703a7bf">

Explanation block: 
<img width="1013" alt="Screenshot 2024-09-04 at 18 57 51" src="https://github.com/user-attachments/assets/ac2deb5d-f187-4a1f-a4b1-187dc9c0f237">

Confirmation: 
<img width="1010" alt="Screenshot 2024-09-04 at 18 34 11" src="https://github.com/user-attachments/assets/2f75c97e-ec57-4a67-934a-d4bf2a29a012">

Confirmation block: 
<img width="1015" alt="Screenshot 2024-09-04 at 18 58 03" src="https://github.com/user-attachments/assets/d7a9ddba-c7a3-4699-95fb-5d744b7ddc59">

Error: 
<img width="1006" alt="Screenshot 2024-09-04 at 18 34 19" src="https://github.com/user-attachments/assets/107ab2b9-62aa-43e5-a167-53d2c9e02a5b">

Error block: 
<img width="1010" alt="Screenshot 2024-09-04 at 18 58 17" src="https://github.com/user-attachments/assets/b39fb8cc-58b6-4163-9c15-fb9d819d7d8c">

Warning: 
<img width="1012" alt="Screenshot 2024-09-04 at 18 34 25" src="https://github.com/user-attachments/assets/ed7ded92-2f51-4d2a-a340-0bcecb7ccecc">

Warning block: 
<img width="1014" alt="Screenshot 2024-09-04 at 18 58 10" src="https://github.com/user-attachments/assets/3a74ec13-0287-4242-b257-7b68625aae22">

System: 
<img width="1010" alt="Screenshot 2024-09-04 at 18 35 53" src="https://github.com/user-attachments/assets/7bb35902-133a-417b-9506-2ce5098001cb">

System block: 
<img width="1012" alt="Screenshot 2024-09-04 at 18 59 02" src="https://github.com/user-attachments/assets/633f66b7-d33d-4859-95d0-0e876829d2f1">


### QA notes

- All notifications should have these updates. Check if you see anything weird happening or if icons are missing on notifications. 
- Please check block notifications as well. I'm not sure if we already have them in the UI but you can follow the steps below to add a test case. 

To test block notifications follow these steps: 
1: Add the html for a block notification to any page. For example: `/nl-kat-coordination/rocky/account/templates/account_detail.html` You can add the provided html (see below) on line 96 right below where the `<p class="explanation">...</p>` ends. 
2: Open the user account page (click on the user icon in the top navigation bar) navigate to "Profile". 
3: You should see you're newly added block notification.
4: To check different types. Change the class name from "explanation to any of the following options: "error" "warning" "system" "confirmation"

```html
<div class="explanation" role="group" aria-label="Toelichting">
    <div>
        <span class="visually-hidden">Toelichting:</span>
        <h2>Lorem ipsum dolor set</h2>
        <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec nec ante a nisi consequat facilisis ac eu velit. Donec luctus metus a lacus dictum, a porta ligula pellentesque. Phasellus ultricies purus id viverra ultrices. Proin sagittis rhoncus tempor. Nam efficitur elit commodo elit feugiat scelerisque. Nullam consectetur felis mi, ac bibendum urna convallis a. Ut maximus volutpat ligula, et accumsan ipsum varius eget. Mauris euismod a augue sit amet condimentum.
        </p>
        <ul>
            <li>Lorem ipsum</li>
            <li>Lorem ipsum</li>
            <li>Lorem ipsum</li>
            <li>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec nec ante a nisi consequat facilisis ac eu velit. <a href="">Lorem ipsum</a> consectetur adipiscing elit.
            </li>
        </ul>
        <button type="button" class="plain">
            Lorem <span aria-hidden="true" class="icon ti-chevron-right"></span>
        </button>
    </div>
</div>
```

### Code Checklist







---

## Checklist for code reviewers:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---

## Checklist for QA:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.
